how-to.v2/use-a-sidecar-app-basic/platform/client/src/provider.ts

```diff
@@ -0,0 +1,217 @@
+import OpenFin from "@openfin/core";
+import type { ChannelClient } from "@openfin/core/src/OpenFin";
+
+let loggingElement: HTMLElement | null;
+let clearLogsBtn: HTMLButtonElement | null;
+let connectToSideCarBtn: HTMLButtonElement | null;
+let dispatchMessageToSideCarBtn: HTMLButtonElement | null;
+let channelClient : ChannelClient;
+
+const SIDECAR_APP_ALIAS = "sidecar-app";
+const SIDECAR_APP_TARGET = "SideCar.App.exe";
+const SIDECAR_APP_SRC = `${ location.origin }/assets/sidecar-app.zip`; 
+/**
+ * Wait for the DOM to have been loaded before we connect the UI elements and initialize the platform.
+ */
+window.addEventListener("DOMContentLoaded", async () => {
+	await initializeDom();
+	await initializePlatform();
+});
+
+/**
+ * Initialize the DOM elements.
+ */
+async function initializeDom(): Promise<void> {
+	loggingElement = document.querySelector("#logging");
+	const loggingContainer: HTMLDivElement | null = document.querySelector("#logging-container");
+
+	if (!loggingElement || !loggingContainer) {
+		return;
+	}
+
+	loggingContainer.style.display = "flex";	
+
+	clearLogsBtn = document.querySelector("#btnClear");
+	clearLogsBtn?.addEventListener("click", () => loggingElement ? loggingElement.textContent = "" : null);
+
+	connectToSideCarBtn = document.querySelector("#btnConnectToSideCarApp");
+	connectToSideCarBtn?.addEventListener("click", async ()=> {
+		await setupSideCarApp();
+		if(connectToSideCarBtn !== null) {
+			connectToSideCarBtn.disabled = true;
+		}
+		if(dispatchMessageToSideCarBtn !== null) {
+			dispatchMessageToSideCarBtn.disabled = false;
+		}
+	});
+
+	dispatchMessageToSideCarBtn = document.querySelector("#btnDispatchMessageToSideCarApp");
+	dispatchMessageToSideCarBtn?.addEventListener("click", dispatchMessageToSideCarApp);
+}
+
+/**
+ * Initialize the Platform.
+ */
+async function initializePlatform(): Promise<void> {
+	try {
+		await fin.Platform.init({ });
+		loggingAddEntry("Platform initialized.");
+	} catch(error) {
+		loggingAddEntry(`There was an error while trying to initialize your platform. Error: ${getErrorMessage(error)}`);		
+	}
+}
+
+/**
+ * Return a message from an error.
+ * @param error The error to inspect and retrieve the message from
+ * @returns The error message
+ */
+function getErrorMessage(error: unknown): string {
+	if (error instanceof Error) return error.message
+	return String(error)
+}
+
+/**
+ * Create and assign a channel client
+ * @returns channel client
+ */
+async function createChannelClient(): Promise<ChannelClient> {
+	const channelName = "sidecar-app";
```

Comment:
Can we make the channel names global consts and the same in the C# app, easier to spot them